node/pkg/watchers/hedera/watcher.go

```diff
@@ -0,0 +1,606 @@
+	// txHash, err := StringToHash(txHashString)
+	// if err != nil {
+	// logger.Error("cannot decode txHashHex", zap.String("txHash", txHashString))
+	// return nil, fmt.Errorf("Could not decode txhash %s", txHashBase.String())
+	// }
+
+	timeStamp := timeStampBase.Int()
+
+	emitter := emitterBase.String()
+	emitterAddr, err := vaa.StringToAddress(emitter)
+	if err != nil {
+		logger.Fatal("failed to unpack emitter address", zap.Error(err))
+		return nil, fmt.Errorf("Emitter field could not be converted for txhash %s, value[%s]", txHashBase.String(), emitter)
+	}
+
+	// Get the other values from the Data value
+	logDataString := logDataBase.String()
+	logDataString = logDataString[2:] // remove the leading 0x
+	logDataBytes, err := hex.DecodeString(logDataString)
+	if err != nil {
+		logger.Fatal("failed to unpack log data", zap.Error(err))
+		return nil, fmt.Errorf("Data field could not be converted for txhash %s, value[%s]", txHashBase.String(), &logDataString)
+	}
+
+	unpackedMsg, err := ethAbi.Unpack("LogMessagePublished", logDataBytes)
+	if err != nil {
+		logger.Fatal("failed to unpack log data", zap.Error(err))
+		return nil, fmt.Errorf("Log Data field could not be unpacked for txhash %s", txHashBase.String())
+	}
+
+	seq := unpackedMsg[0].(uint64)
+	nonce := unpackedMsg[1].(uint32)
+	payload := unpackedMsg[2].([]byte)
+	cLevel := unpackedMsg[3].(uint8)
+	logger.Info("unpackedMsg", zap.Int("Length of unpackedMsg", len(unpackedMsg)), zap.Uint64("0", seq), zap.Uint32("2", nonce), zap.Uint8("4", cLevel))
+	var chainID vaa.ChainID
+
+	messagePublication := &common.MessagePublication{
+		TxHash:           txHash,                  // In log
+		Timestamp:        time.Unix(timeStamp, 0), // In log
+		Nonce:            nonce,                   // In log data
+		Sequence:         seq,                     // In log data
+		EmitterChain:     chainID,                 // Don't know where to get this
+		EmitterAddress:   emitterAddr,             // In log
+		Payload:          payload,                 // In log data
+		ConsistencyLevel: cLevel,                  // In log data
+	}
+
+	logger.Info("messagePublication",
+		zap.Stringer("txHash", txHash),
+		zap.Int64("timestamp", timeStamp),
+		zap.Uint32("nonce", nonce),
+		zap.Uint64("sequence", seq),
+		zap.Stringer("EmitterAddr", emitterAddr),
+		zap.Uint8("ConsistenceLevel", cLevel),
+	)
+
+	return messagePublication, nil
+}
+
+func EventsToMessagePublications(contract string, txHash string, events []gjson.Result, logger *zap.Logger, chainID vaa.ChainID, contractAddressKey string) []*common.MessagePublication {
+	networkName := vaa.ChainID(chainID).String()
+	msgs := make([]*common.MessagePublication, 0, len(events))
+	for _, event := range events {
+		if !event.IsObject() {
+			logger.Warn("event is invalid", zap.String("network", networkName), zap.String("tx_hash", txHash), zap.String("event", event.String()))
+			continue
+		}
+		eventType := gjson.Get(event.String(), "type")
```

Comment:
can't you just do: gjson.Get("type")... why are you converting it to a string first which causes it to get re-parsed?

Reply:
event.Get("type")

Reply:
Removed code